building.md

```diff
@@ -0,0 +1,86 @@
+# Build Vita3K
+
+## Windows (Visual Studio)
+
+- Install Visual Studio 2017 or 2019 and choose to install `Desktop development with C++` (as shown in this [image](https://i.imgur.com/bkY15Oh.png)). You will get compiler and `cmake` required for building.
+
+- Install `git` to `clone` the project. Download and install `git` from [here](https://git-scm.com).
+
+- Clone this repo.
+
+```cmd
+git clone --recursive https://github.com/Vita3K/Vita3K
+cd Vita3K
+```
+
+- Run `gen-windows.bat` to create Visual Studio project in `build-windows` directory.
+
+```cmd
+.\gen-windows
+```
+
+- Open the project generated in the `build-windows` directory.
+
+- Build and run it.
+
+## macOS (Xcode)
+
+- Install Xcode at App Store.
+
+- Install [`brew`](https://brew.sh).
+
+- Install dependencies with `brew`.
+
+```sh
+brew install git cmake boost
+```
+
+- Clone this repo.
+
+```sh
+git clone --recursive https://github.com/Vita3K/Vita3K
+cd Vita3K
+```
+
+- Generate Xcode project.
+
+```
+./gen-macos.sh
+```
+
+- Open Xcode project `Vita3K.xcodeproj` generated in `build-macos` directory.
+
+- When prompted to create schemes, create one for the `emulator` target only. The project builds many targets, so it will make your life easier if you create schemes as needed.
+
+- Build.
+
+### Linux (Debian)
+
+- Install dependencies.
+
+```sh
+sudo apt install git cmake libboost-all-dev libsdl2-dev
+```
+
+- Clone this repo.
+
+```sh
+git clone --recursive https://github.com/Vita3K/Vita3K
+cd Vita3K
+```
+
+- Build the project.
+
+```sh
+./gen-linux.sh
+cd build-linux
+make -j$(nproc)
+```
+
+#### Note: If Unicorn can't find Python, use `make UNICORN_QEMU_FLAGS="--python=/usr/bin/python2"` or similar to point it to your installation. `make` works after you do this once.
+
+## Note
+
+- After cloning or checking out a branch, you should always update submodules.
+
+`git submodule update --init --recursive`
```

Comment:
Since this applies to everything and you should do this before building, maybe put this on the top?

Reply:
I also think this is important but putting it first doesn't make much sense. This should be applied when someone is actively changing sub-modules or generating untracked files in sub-modules.

It just not make much sense for putting it on top. I think we should have a `CONTRIBUTING.md` later and move technical notes to there.

Reply:
`git clone --recursive` will get the job done for one-time visitors.